iceberg-rust/src/view/transaction/operation.rs

```diff
@@ -35,19 +35,60 @@ pub enum Operation {
     UpdateProperties(Vec<(String, String)>),
 }
 
+// Tries to preserve dialect order
+fn upsert_representation(
+    current_representations: &[ViewRepresentation],
+    new_representation: ViewRepresentation,
+) -> Vec<ViewRepresentation> {
+    let ViewRepresentation::Sql {
+        dialect: new_dialect,
+        ..
+    } = &new_representation;
+    let mut updated = false;
+    let mut representations: Vec<ViewRepresentation> = current_representations
+        .iter()
+        .map(
+            |current_representation @ ViewRepresentation::Sql { dialect, .. }| {
+                if dialect == new_dialect {
+                    updated = true;
+                    new_representation.clone()
+                } else {
+                    current_representation.clone()
+                }
+            },
+        )
+        .collect();
+    if !updated {
+        representations.push(new_representation);
+    }
+    representations
+}
+
+fn upsert_representations(
+    current_representations: &[ViewRepresentation],
+    new_representations: &[ViewRepresentation],
+) -> Vec<ViewRepresentation> {
+    let mut representations: Vec<ViewRepresentation> = current_representations.into();
+    for r in new_representations {
+        representations = upsert_representation(&representations, r.clone());
+    }
```

Comment:
I really dislike such naming. For such a small code block is easy to see what it is, but if for-block grows, it becomes annoying to figure it out what the single letter variable is. 

The `mut` avoids cloning and rebuilding `representations` for each new representation. 

```suggestion
    for representation in new_representations {
        representations = upsert_representation(&mut representations, representation.clone());
    }
```